mava/utils/jax_training_utils.py

```diff
@@ -44,34 +44,53 @@ def construct_norm_axes_list(
     start_axes: int, axes_list: List[Any], obs_shape: Tuple
 ) -> Tuple[slice, ...]:
     """Construt a list of Tuples containing the features on which to apply normalisation
-
-    start_axes (int) --- default axes from which to start
-    axes_list (List[int]) --- user specified
-    obs_shape (Tuple) --- observations shape
+    Args:
+        start_axes (int) --- default axes from which to start
+        axes_list (List[int]) --- user specified
+        obs_shape (Tuple) --- observations shape
 
     Returns:
         axes_list: a tuple to be used with np.r_
+
+    The start_axes is 0 unless we use a wrapper like concat_agent_id
+    or concat_previous_actions which add one hot encorded vectors
+    at the start of the array.
+    axes_list corresponds to user specified axes we want to normalise
+    We aussume the user does not consider contenation when specifying axes_list.
+    If axes_list is empty then we nornmalise all the axes
+    axes_list can contain single values or lists and tuples which corresponds to
+    start and end values of axes slices. eg. [1, 2, [4,7], (9,15)]
+    If start_axes is different from 0 then we need offset all
+    the enteries in axes_list by the start_axes
+    For the axes_list [1, 2, [4,7], (9,15)] with start_axes = 0
+    output is tuple([slice(1,2), slice(2,3), slice(4,7), slice(9,15)]).
+    if axes_list = [] and start_axes = 0
+    output is tuple([slice(0, 15)]) assuming obs_shape = (15,)
     """
 
     if len(axes_list) == 0:
-        return_list = [slice(start_axes, obs_shape[0])]
+        return tuple([slice(start_axes, obs_shape[0])])
     else:
         return_list = []
-        indices = []
+        starts = []
+        ends = []
         for x in axes_list:
             if type(x) == tuple or type(x) == list:
                 element = slice(x[0] + start_axes, x[1] + start_axes)
-                indices.append(x[1] + start_axes)
+                starts.append(x[0] + start_axes)
+                ends.append(x[1] + start_axes)
```

Comment:
As discussed offline, you can keep or remove this, up to you

Reply:
I have decided to keep it. It just few lines of code and we never know if we may need it.